aspnetcore/fundamentals/openapi/include-metadata.md

```diff
@@ -520,6 +522,79 @@ Note that object and dynamic types have _no_ type defined in the OpenAPI because
 
 The `type` and `format` can also be set with a [Schema Transformer](xref:fundamentals/openapi/customize-openapi#use-schema-transformers). For example, you may want the `format` of decimal types to be `decimal` instead of `double`.
 
+:::moniker-end
+
+:::moniker range=">= aspnetcore-10.0"
+
+#### Numeric types
+
+The JSON Schema library maps standard C# numeric types to OpenAPI `type` and `format` based on the
+<xref:System.Text.Json.JsonSerializerOptions.NumberHandling> property of the <xref:System.Text.Json.JsonSerializerOptions>
+used in the app. In ASP.NET Core Web API apps, the default value of this property is `JsonNumberHandling.AllowReadingFromString`.
+
+When the <xref:System.Text.Json.JsonSerializerOptions.NumberHandling> property is set to `JsonNumberHandling.AllowReadingFromString`, the numeric types are mapped as follows:
+
+| C# Type        | OpenAPI `type`   | OpenAPI `format` | Other assertions               |
+| -------------- | ---------------- | ---------------- | ------------------------------ |
+| int            | [integer,string] | int32            | pattern `<digits>` |
+| long           | [integer,string] | int64            | pattern `<digits>` |
+| short          | [integer,string] | int16            | pattern `<digits>` |
+| byte           | [integer,string] | uint8            | pattern `<digits>` |
+| float          | [number,string]  | float            | pattern `<digits with decimal >` |
+| double         | [number,string]  | double           | pattern `<digits with decimal >` |
+| decimal        | [number,string]  | double           | pattern `<digits with decimal >` |
+
+<!--
+| int            | [integer,string] | int32            | pattern `<digits>` |  pattern: `"^-?(?:0|[1-9]\\d*)$"`
+| long           | [integer,string] | int64            | pattern `<digits>` |                                 
+| short          | [integer,string] | int16            | pattern `<digits>` |                                  
+| byte           | [integer,string] | uint8            | pattern `<digits>` |                                  
+| float          | [number,string]  | float            | pattern `<digits with decimal >` |   pattern: `"^-?(?:0|[1-9]\\d*)(?:\\.\\d+)?(?:[eE][+-]?\\d+)?$"`
+| double         | [number,string]  | double           | pattern `<digits with decimal >` |                                  
+| decimal        | [number,string]  | double           | pattern `<digits with decimal >` |   pattern: `"^-?(?:0|[1-9]\\d*)(?:\\.\\d+)?$"`
```

Comment:
Took me awhile to figure that build error. See the `|` in the pattern? Even commented out, the parser picks up the column separator before stripping comment - error
even with ``` back ticks, same column error. Very strange

<img width="272" alt="image" src="https://github.com/user-attachments/assets/2612529f-9570-4569-9e8d-fd7bb42febe7" />

The other build error took a bit too, you had `:: moniker-end`, only two `:`, and 3 required. I've never made that mistake - every, because I'm dyslexic and can't type. I copy/paste.

Reply:
@tdykstra @guardrex strange you can't have comments after a row that include a pipe `|` - even in a comment, even `code fenced`

Reply:
Oh @Rick-Anderson ... so sorry! I think I force-pushed and erased some of your changes by accident. I should have been more careful. I will try to recover.

Reply:
Okay ... I think I restored your version. I will be more careful from now on.

Reply:
> Oh @Rick-Anderson ... so sorry! I think I force-pushed and erased some of your changes by accident. I should have been more careful. I will try to recover.

I should have made clear I pushed some commit, sorry. I thought you were finished.